### PR TITLE
Add 3 new tools to cpt.yaml

### PR DESCRIPTION
**Pull request by jasonjgill**
Adds 3 new CPT phage tools, deletes 2 deprecated CPT tools, adds 3 new useful phage tools from the toolshed.

### CODE REVIEW

File: cpt.yaml [hunk 1]
```diff
@@ -64,18 +64,10 @@ tools:
     owner: cpt
     tool_panel_section_label: Annotation
 
-  - name: cpt_intersect_adjacent
```
Comment (bgruening) [inline cpt.yaml:67]: We can not remove tools, they need to stay for reproducibility ... we could hide them if this is useful.

Please let them listed here and add a comment that they are deprecated.

File: cpt.yaml [hunk 2]
```diff
@@ -127,3 +119,27 @@ tools:
...
+  - name: cpt_prophage_relatedness
+    owner: cpt
+    tool_panel_section_label: Annotation
+
+  - name: cpt_intersect_adj
+    owner: cpt
+    tool_panel_section_label: Annotation
+
+  - name: phageterm
+    owner: mmonot
```
Comment (bgruening) [inline cpt.yaml:136]: We need to review those tools before we can merge them.

Comment (bgruening) [reply cpt.yaml:136]: Can you point me maybe to the github repo of those tools?

**Comment by jasonjgill** [comment]
Hi Bjoern,

Here are the repos for our tools:

cpt_protein_blast_grouping :: https://github.com/TAMU-CPT/CPT-ToolshedSource/tree/main/cpt_protein_blast_grouping
cpt_prophage_relatedness :: https://github.com/TAMU-CPT/CPT-ToolshedSource/tree/main/cpt_prophage_relatedness
cpt_intersect_adj :: https://github.com/TAMU-CPT/CPT-ToolshedSource/tree/main/cpt_intersect_adj

The other three tools are from other authors but would also be useful for Phage Galaxy. These tools all worked on Curtis’ Galaxy testbed. Let us know if you need any edits. I will also revert the deletions of those two other tools, should I just add a ## comment line under each of those to state they are deprecated?

Thanks,

Jason.

File: cpt.yaml (hunk 2 above)
Comment (bgruening) [reply cpt.yaml:136]: https://github.com/avilella/phageterm  <- is a fork 

The only reference on github is https://github.com/TAMU-CPT/galaxy-tools/blob/c43f5a7706fb64615b7c050a5a9e5fc20ca33873/tool_conf.xml.old#L76 ... any reason you have hidden it on your instance?


I think I found the latest version https://gitlab.pasteur.fr/vlegrand/ptv/-/tree/master

But the Galaxy wrapper is 6 years old and has not been updated. I could not find the source on github, the source in the TS is insecure and needs updates.

File: cpt.yaml [hunk 3]
```diff
@@ -127,3 +119,27 @@ tools:
...
+    tool_panel_section_label: Annotation
+
+  - name: phagetermvirome
+    owner: bioit_sciensano
+    tool_panel_section_label: Annotation
```
Comment (bgruening) [inline cpt.yaml:141]: This one also seems to be https://gitlab.pasteur.fr/vlegrand/ptv/-/tree/master

> The previous PhageTerm program (single phage analysis only) is still available at https://sourceforge.net/projects/phageterm/ (for versions <3.0.0)

But also no Galaxy wrapper in github/gitlab.

The wrapper in the TS looks ok, a few things can be improved.

File: cpt.yaml [hunk 4]
```diff
@@ -127,3 +119,27 @@ tools:
...
+    
+  - name: defense_finder
+    owner: rplanel
```
Comment (bgruening) [inline cpt.yaml:144]: That one looks ok, I found the wrapper and it looks like a good quality.

**Review by bgruening — COMMENTED** [review]
@jasonjgill I tried to find the source of the tools, see my comments inline.

I'm not concerned about functionality. I trust you all when you say those tools are useful. My problem is with sustainability and security. When those wrappers are not maintained over time, do not have tests, or do not adhere to basic security needs of a public Galaxy server, I have problems to install them.

Thats why I recommend to contribute your tools to IUC as well. There we have a review process, rigorous testing and bots that help us maintaining the warppers.

**Comment by jasonjgill** [comment]
Hi Bjoern,

I have cc'd Curtis who can address issues in our repo. The third-party tools are outside of our control, I guess all we could do would be to try to update them and see if they will merge our changes. We can drop tools from the PR for now if needed, the priority for us are the 3 CPT tools. Getting into IUC is our eventual plan, it's mostly an issue of available resources. We have a new worker we might be able to task with this.

Thanks,

Jason.

Jason Gill
Associate Professor
Dept. of Animal Science
Texas A&M University

Sent via mobile.

On Sep 1, 2024 9:45 AM, Björn Grüning ***@***.***> wrote:
@ bgruening commented on this pull request. @jasonjgill I tried to find the source of the tools, see my comments inline. I'm not concerned about functionality. I trust you all when you say those tools are useful. My problem is with sustainability
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.

ZjQcmQRYFpfptBannerEnd

@bgruening commented on this pull request.

@jasonjgill<https://urldefense.com/v3/__https://github.com/jasonjgill__;!!KwNVnqRv!H1w7o4CxfBrt_85mrD2AgxWW6VteWFZWpwkccbEZnjYSlRjUhlEBQVJdmuvgQRlAbNG3OQ3g0qbA0x_18nqNRGjAnt4$> I tried to find the source of the tools, see my comments inline.

I'm not concerned about functionality. I trust you all when you say those tools are useful. My problem is with sustainability and security. When those wrappers are not maintained over time, do not have tests, or do not adhere to basic security needs of a public Galaxy server, I have problems to install them.

Thats why I recommend to contribute your tools to IUC as well. There we have a review process, rigorous testing and bots that help us maintaining the warppers.

________________________________

In cpt.yaml<https://urldefense.com/v3/__https://github.com/usegalaxy-eu/usegalaxy-eu-tools/pull/776*discussion_r1740122805__;Iw!!KwNVnqRv!H1w7o4CxfBrt_85mrD2AgxWW6VteWFZWpwkccbEZnjYSlRjUhlEBQVJdmuvgQRlAbNG3OQ3g0qbA0x_18nqN8Bk7uV0$>:

> +
+  - name: cpt_prophage_relatedness
+    owner: cpt
+    tool_panel_section_label: Annotation
+
+  - name: cpt_intersect_adj
+    owner: cpt
+    tool_panel_section_label: Annotation
+
+  - name: phageterm
+    owner: mmonot
+    tool_panel_section_label: Annotation
+
+  - name: phagetermvirome
+    owner: bioit_sciensano
+    tool_panel_section_label: Annotation


This one also seems to be https://gitlab.pasteur.fr/vlegrand/ptv/-/tree/master<https://urldefense.com/v3/__https://gitlab.pasteur.fr/vlegrand/ptv/-/tree/master__;!!KwNVnqRv!H1w7o4CxfBrt_85mrD2AgxWW6VteWFZWpwkccbEZnjYSlRjUhlEBQVJdmuvgQRlAbNG3OQ3g0qbA0x_18nqNAkdnVL0$>

The previous PhageTerm program (single phage analysis only) is still available at https://sourceforge.net/projects/phageterm/<https://urldefense.com/v3/__https://sourceforge.net/projects/phageterm/__;!!KwNVnqRv!H1w7o4CxfBrt_85mrD2AgxWW6VteWFZWpwkccbEZnjYSlRjUhlEBQVJdmuvgQRlAbNG3OQ3g0qbA0x_18nqNNtyzbM8$> (for versions <3.0.0)

But also no Galaxy wrapper in github/gitlab.

The wrapper in the TS looks ok, a few things can be improved.

________________________________

In cpt.yaml<https://urldefense.com/v3/__https://github.com/usegalaxy-eu/usegalaxy-eu-tools/pull/776*discussion_r1740126241__;Iw!!KwNVnqRv!H1w7o4CxfBrt_85mrD2AgxWW6VteWFZWpwkccbEZnjYSlRjUhlEBQVJdmuvgQRlAbNG3OQ3g0qbA0x_18nqNxxrkjIM$>:

> +    tool_panel_section_label: Annotation
+
+  - name: cpt_intersect_adj
+    owner: cpt
+    tool_panel_section_label: Annotation
+
+  - name: phageterm
+    owner: mmonot
+    tool_panel_section_label: Annotation
+
+  - name: phagetermvirome
+    owner: bioit_sciensano
+    tool_panel_section_label: Annotation
+
+  - name: defense_finder
+    owner: rplanel


That one looks ok, I found the wrapper and it looks like a good quality.

________________________________

In cpt.yaml<https://urldefense.com/v3/__https://github.com/usegalaxy-eu/usegalaxy-eu-tools/pull/776*discussion_r1740122799__;Iw!!KwNVnqRv!H1w7o4CxfBrt_85mrD2AgxWW6VteWFZWpwkccbEZnjYSlRjUhlEBQVJdmuvgQRlAbNG3OQ3g0qbA0x_18nqN_cRG9WQ$>:

> @@ -127,3 +119,27 @@ tools:
   - name: cpt_fasta_translate
     owner: cpt
     tool_panel_section_label: Annotation
+
+  - name: cpt_protein_blast_grouping
+    owner: cpt
+    tool_panel_section_label: Annotation
+
+  - name: cpt_prophage_relatedness
+    owner: cpt
+    tool_panel_section_label: Annotation
+
+  - name: cpt_intersect_adj
+    owner: cpt
+    tool_panel_section_label: Annotation
+
+  - name: phageterm
+    owner: mmonot


https://github.com/avilella/phageterm<https://urldefense.com/v3/__https://github.com/avilella/phageterm__;!!KwNVnqRv!H1w7o4CxfBrt_85mrD2AgxWW6VteWFZWpwkccbEZnjYSlRjUhlEBQVJdmuvgQRlAbNG3OQ3g0qbA0x_18nqNSia5iMs$> <- is a fork

The only reference on github is https://github.com/TAMU-CPT/galaxy-tools/blob/c43f5a7706fb64615b7c050a5a9e5fc20ca33873/tool_conf.xml.old#L76<https://urldefense.com/v3/__https://github.com/TAMU-CPT/galaxy-tools/blob/c43f5a7706fb64615b7c050a5a9e5fc20ca33873/tool_conf.xml.old*L76__;Iw!!KwNVnqRv!H1w7o4CxfBrt_85mrD2AgxWW6VteWFZWpwkccbEZnjYSlRjUhlEBQVJdmuvgQRlAbNG3OQ3g0qbA0x_18nqNN2uONbA$> ... any reason you have hidden it on your instance?

I think I found the latest version https://gitlab.pasteur.fr/vlegrand/ptv/-/tree/master<https://urldefense.com/v3/__https://gitlab.pasteur.fr/vlegrand/ptv/-/tree/master__;!!KwNVnqRv!H1w7o4CxfBrt_85mrD2AgxWW6VteWFZWpwkccbEZnjYSlRjUhlEBQVJdmuvgQRlAbNG3OQ3g0qbA0x_18nqNAkdnVL0$>

But the Galaxy wrapper is 6 years old and has not been updated. I could not find the source on github, the source in the TS is insecure and needs updates.

**Comment by jasonjgill** [comment]
Hi Bjoern,

If we drop phageterm, phagetermvirome and defense_finder from the PR, are you OK with installing the other tools? The CPT tools are the ones we need the most.

Thanks,

Jason.
________________________________

**Comment by jasonjgill** [comment]
Made a new commit to add back the 2 deleted tools (with comments) and removed the third-party tools. Net change now is just the 3 new CPT tools.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
I have hidden the 2 deprecated tools ... :)

**Pull request merged**